source/common/upstream/outlier_detection_impl.cc

```diff
@@ -471,8 +475,28 @@ void DetectorImpl::ejectHost(HostSharedPtr host,
           runtime_.snapshot().getInteger(BaseEjectionTimeMsRuntime, config_.baseEjectionTimeMs()));
       const std::chrono::milliseconds max_eject_time = std::chrono::milliseconds(
           runtime_.snapshot().getInteger(MaxEjectionTimeMsRuntime, config_.maxEjectionTimeMs()));
-      if ((host_monitors_[host]->ejectTimeBackoff() * base_eject_time) <
-          (max_eject_time + base_eject_time)) {
+      const std::chrono::milliseconds max_eject_time_jitter =
+          std::chrono::milliseconds(runtime_.snapshot().getInteger(
+              MaxEjectionTimeJitterMsRuntime, config_.maxEjectionTimeJitterMs()));
+
+      // Generate random jitter so that not all hosts uneject at the same time,
+      // which could possibly generate a connection storm
+
+      // std::chrono::milliseconds receives a signed integer as input, so we ensure that our random
+      // random value is in the range [0, std::numeric_limits<long>::max())
+      long random_long =
+          static_cast<long>(random_generator_() % (std::numeric_limits<long>::max()));
+      // cap the jitter at max_eject_time_jitter
```

Comment:
nit: Comments start with capital letter and end with '.'

Reply:
Fixed